### PR TITLE
Add remote rpc support

### PR DESCRIPTION
**Pull request by eddiewang**
Never written rust before, so might need some polish here before it's merge ready.
Mainly quality of life improvements for users that aren't using a localhost Bitcoin node.

- Adds rpc_user and rpc_pass arguments to the binary
- Uses rpc_user / rpc_pass if arguments exist, otherwise defaults back to cookie mode
- Ensures all wallet commands properly pass the /wallet/<name> even if the rpc_url is set

### CODE REVIEW

**Comment by eddiewang** [comment]
Right now, the logic will default to cookie unless both arguments are provided.
My understanding of how `.is_none()` works is that if the value is set, even if it's to an empty string `""`, it should still return `False`. 

If my assumption is correct, I think the desired state is to explicitly pass in both the user and pass in order to activate this mode, defaulting to the cookie approach otherwise.

File: src/index.rs [hunk 1]
```diff
@@ -133,15 +133,31 @@ impl<T> BitcoinCoreRpcResultExt<T> for Result<T, bitcoincore_rpc::Error> {
...
+      Auth::UserPass(ruser.clone(), rpass.clone())
+    } else {
+      log::info!(
+        "Connecting to Bitcoin Core RPC server at {rpc_url} using credentials from `{}`",
+        cookie_file_path.display()
+      );
+      auth
+    };
+
+
```
Comment (coconutpalms) [inline src/index.rs:160]: I think you need to make similar changes around L161 of options.rs

File: src/index.rs [hunk 2]
```diff
@@ -134,15 +134,31 @@ impl<T> BitcoinCoreRpcResultExt<T> for Result<T, bitcoincore_rpc::Error> {
...
-    );
 
+    // by default we use a cookie file
     let auth = Auth::CookieFile(cookie_file);
 
+    // if options.rpc_pass and rpc_user is not empty, use Auth::UserPass
+    let auth = if !rpc_pass.is_none() && !rpc_user.is_none() {
+      let ruser = rpc_user.unwrap();
+      let rpass = rpc_pass.unwrap();
+      log::info!("Using RPC credentials from command line {ruser}");
```
Comment (AngusP) [inline src/index.rs:150]: `rpc_pass` and `rpc_user` are `Option<String>`, which you can get at in a nicer way than checking `! is_none()` then calling `unwrap()` (calling unwrap is generally (strongly) discouraged), by using [`if let`](https://doc.rust-lang.org/rust-by-example/flow_control/if_let.html) and [destructuring](https://doc.rust-lang.org/rust-by-example/flow_control/match/destructuring/destructure_tuple.html):

```rust
let number = Some(7);
let another = Some(12);
let missing: Option<i32> = None;

if let (Some(i), Some(j)) = (number, another) {
    println!("Matched {:?} and {:?}!", i, j);
}
// ^ Prints "Matched 7 and 12!"

if let (Some(i), Some(j)) = (number, missing) {
    println!("Matched {:?} and {:?}!", i, j);
} else {
    println!("Didn't match");
}
// ^ Prints "Didn't match"
```

so something like

```suggestion
    if let (Some(rpass), Some(ruser))  = (rpc_pass, rpc_user) {
      log::info!("Using RPC credentials from command line {ruser}");
```

(Caveat I'm also a Rust noob so if the compiler complains, sorry!)

**Comment by AngusP** [comment]
Also note #1527 was trying to add this as well

**Comment by raphjaph** [comment]
We added this check out PR #1527

**Pull request closed without merging**